fedimint-api/src/db/mod.rs

```diff
@@ -169,6 +169,93 @@ impl Drop for CommitTracker {
     }
 }
 
+pub struct ModuleDatabaseTransaction<'a, 'b: 'a> {
+    tx: &'a mut DatabaseTransaction<'b>,
+    module_prefix: Vec<u8>,
+}
+
+impl<'a, 'b> ModuleDatabaseTransaction<'a, 'b> {
+    pub fn new(
+        dbtx: &'a mut DatabaseTransaction<'b>,
+        module_prefix: u16,
+    ) -> ModuleDatabaseTransaction<'a, 'b> {
+        ModuleDatabaseTransaction {
+            tx: dbtx,
+            module_prefix: vec![(module_prefix >> 8) as u8, (module_prefix & 0xff) as u8],
+        }
+    }
+
+    pub fn new_without_prefix(
+        dbtx: &'a mut DatabaseTransaction<'b>,
+    ) -> ModuleDatabaseTransaction<'a, 'b> {
+        ModuleDatabaseTransaction {
+            tx: dbtx,
+            module_prefix: vec![],
+        }
+    }
+
+    pub async fn get_value<K>(&mut self, key: &K) -> Result<Option<K::Value>>
```

Comment:
We should not be implementing these functions, and only `raw` ones, and call into the `self.tx` only on raw ones as well.

Reply:
So we prefix the prefix, call `self.tx.<the_same_raw_method>`, strip the prefix, and return back.

Reply:
`raw_find_by_prefix` might be a bit of a problem, so we might need to build in suffix stripping right into `PrefixIter`, to avoid another layer of dynamic polymorphism, just to substitute it with a prefix-stripping implementation..

Reply:
This should be fixed now. I ended up doing the pre-pending and stripping of the prefix in raw_find_by_prefix of IsolatedDatabaseTransaction